core/iwasm/aot/aot_runtime.c

```diff
@@ -1058,6 +1073,16 @@ aot_get_default_memory(AOTModuleInstance *module_inst)
         return NULL;
 }
 
+AOTMemoryInstance *
+aot_get_memory_with_index(AOTModuleInstance *module_inst, uint32 index)
+{
+    bh_assert(index < module_inst->memory_count);
```

Comment:
I think if we want to make the `aot_get_memory_with_index` export API, we need to check the index in the below if statement rather than simply assert. In the past, the loader will basically check and make sure the index is not out of bounds,  if now the user will provide the index, we better to further check it. 

The same rationale applies to `wasm_get_memory_with_index`